src/System.Runtime.InteropServices.RuntimeInformation/src/RuntimeInformation.Windows.cs

```diff
@@ -1,13 +1,83 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Diagnostics;
+
 namespace System.Runtime.InteropServices
 {
-    public static class RuntimeInformation
+    public static partial class RuntimeInformation
     {
+        private static string s_osDebugName = null;
+        private static Architecture? s_osArch = null;
+
         public static bool IsOSPlatform(OSPlatform osPlatform)
         {
             return OSPlatform.Windows == osPlatform;
         }
+
+        public static string OSDescription
+        {
+            get
+            {
+                if (null == s_osDebugName)
+                {
+                    string type = Interop.mincore.IsWindowsServer() ? "Server" : "Client";
+                    string version = "Unknown";
+
+                    if (Interop.mincore.IsWindows10OrGreater())
+                        version = "10";
+                    else if (Interop.mincore.IsWindows8Point1OrGreater())
+                        version = "8.1";
+                    else if (Interop.mincore.IsWindows8OrGreater())
+                        version = "8";
+                    else if (Interop.mincore.IsWindows7OrGreater())
+                        version = "7";
+                    else if (Interop.mincore.IsWindowsVistaOrGreater())
+                        version = "Vista";
+                    else if (Interop.mincore.IsWindowsXPOrGreater())
+                        version = "XP";
+
+                    s_osDebugName = string.Format("Windows {0} {1} Or Greater", type, version);
+                }
+
+                return s_osDebugName;
+            }
+        }
+
+        public static Architecture OSArchitecture
+        {
+            get
+            {
+                if (null == s_osArch)
+                {
+                    Interop.mincore.SYSTEM_INFO sysInfo;
+                    Interop.mincore.GetNativeSystemInfo(out sysInfo);
+
+                    if ((ushort)Interop.mincore.ProcessorArchitecture.Processor_Architecture_ARM == sysInfo.wProcessorArchitecture)
+                    {
+                        s_osArch = Architecture.Arm;
+                    }
+                    else if (s_is64BitProcess)
+                    {
+                        s_osArch = Architecture.x64;
+                    }
+                    else
+                    {
+                        bool isWow64 = false;
+                        if (Interop.mincore.IsWow64Process(Process.GetCurrentProcess().SafeHandle, ref isWow64) && isWow64)
```

Comment:
Call GetSystemInfo instead of GetNativeSystemInfo to avoid the guess work about Wow - the call to IsWow64Process should not be needed.